apps/docs-snippets/src/utils.ts

```diff
@@ -46,10 +46,7 @@ export const getTestWallet = async (seedQuantities?: CoinQuantityLike[]) => {
   await genesisWallet.fund(request, requiredQuantities, minFee);
 
   // execute the transaction, transferring resources to the test wallet
-  const response = await genesisWallet.sendTransaction(request);
-
-  // wait for the transaction to be confirmed
-  await response.wait();
+  await genesisWallet.sendTransaction(request, { awaitExecution: true });
```

Comment:
`awaitExecution` feels like an implementation detail that is not important to the consumer, and only really to the client. Is the plan to have this completely replace the current submit and retry approach?

Reply:
No, we want to allow the user to be able to await the execution on the node but also to just submit the transaction and call `TransactionResponse.waitForResult` whenever they want. This PR is only introducing `submitAndAwait` and applying it to some places as default (e.g. `BaseInvocationScope.call`) and allowing it to be used in other places (e.g. `Account.transfer`), keeping in mind to not introduce any breaking change.
There are other discussions in circulation ([the comments on this issue](https://github.com/FuelLabs/fuels-ts/pull/1615#discussion_r1447587623), #1588) that will undoubtedly lead to some restructuring and more clarity when to use which feature. Now we're gaining a quick win by utilizing the existence of this endpoint.